eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/persistence/DbsnpSubmittedVariantAccessioningDatabaseService.java

```diff
@@ -62,4 +74,62 @@ public DbsnpSubmittedVariantAccessioningDatabaseService(DbsnpSubmittedVariantAcc
+        List<DbsnpSubmittedVariantEntity> entities = this.repository.findByAccession(accession);
+        this.checkAccessionIsActive(entities, accession);
+        return this.toAccessionWrapper(entities);
+    }
+
+    private void checkAccessionIsActive(List<DbsnpSubmittedVariantEntity> entities, Long accession) throws AccessionDoesNotExistException, AccessionMergedException, AccessionDeprecatedException {
+        if (entities == null || entities.isEmpty()) {
+            this.checkAccessionNotMergedOrDeprecated(accession);
+        }
+    }
+
+    /**
+     * dbSNP submitted variants can be "updated" and "merged" at the same time. Give priority to the "merge" events. No more
+     * than 1 "merge" events will be present.
+     */
+    private void checkAccessionNotMergedOrDeprecated(Long accession)
+            throws AccessionDoesNotExistException, AccessionMergedException, AccessionDeprecatedException {
+        List<? extends IEvent<ISubmittedVariant, Long>> events = inactiveService.getEvents(accession);
+        Optional<? extends IEvent<ISubmittedVariant, Long>> mergedEvent = events.stream()
+                                                                                .filter(e -> EventType.MERGED.equals(
+                                                                                        e.getEventType()))
+                                                                                .findFirst();
+        if (mergedEvent.isPresent()) {
+            throw new AccessionMergedException(accession.toString(), mergedEvent.get().getMergedInto().toString());
+        } else if (events.stream().map(IEvent::getEventType).anyMatch(EventType.DEPRECATED::equals)) {
+            throw new AccessionDeprecatedException(accession.toString());
+        }
+    }
+
+    private AccessionVersionsWrapper<ISubmittedVariant, String, Long> toAccessionWrapper(List<DbsnpSubmittedVariantEntity> entities) {
+        List<AccessionWrapper<ISubmittedVariant, String, Long>> models = entities.stream()
+                                                                                 .map(this::toModelWrapper)
+                                                                                 .collect(Collectors.toList());
+        return new AccessionVersionsWrapper<>(models);
+    }
+
+    @Override
+    public AccessionWrapper<ISubmittedVariant, String, Long> findLastVersionByAccession(Long accession)
+            throws AccessionDoesNotExistException, AccessionMergedException, AccessionDeprecatedException {
+        List<DbsnpSubmittedVariantEntity> entities = this.repository.findByAccession(accession);
+        this.checkAccessionIsActive(entities, accession);
+        return this.toModelWrapper(this.filterOldVersions(entities));
+    }
+
+    private DbsnpSubmittedVariantEntity filterOldVersions(List<DbsnpSubmittedVariantEntity> accessionedElements) {
```

Comment:
Suggest renaming to "filterOutOldVersions"

Reply:
renamed to `getNewestVersion`